### PR TITLE
Cc UI structure

### PR DESCRIPTION
**Pull request by olivena**
This branch is about folder and file structure in CC UI, I dont know why the branch name did not change here, since it changed in gitHub desktop. I added information about the files in the README -file for you to get better understanding of what a file/folder is for. The reason that there are so many folders is to keep it tidyier and that all items/css/etc used in the tsx can be placed in the same folder, so they are easier to find.

This has been made as modeled in figma:
https://www.figma.com/file/eAGHe7brWxyw8W80xRHlXE/Connected-Circularity-team-library?node-id=432%3A3911

### CODE REVIEW

**Comment by rvsape** [comment]
the ui - folder and it's content seems unnecessary, is it needed?

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,202 @@
+{
```
Comment (rvsape) [inline package-lock.json]: I think this lock file is misplaced?

**Comment by rvsape** [comment]
the pages-folder is bit buried under the components folder, I suggest removing it directly under src

File: cc-ui/src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
 
 function App() {
   return (
     <div className="App">
-      <h1>Welcome to Connected Circularity!</h1>
-    </div>
-  )
+      <Routes>
+        <Route path='/' element={<Login/>} />
+        <Route path='/Home' element={<Home/>} />
```
Comment (rvsape) [inline cc-ui/src/App.tsx]: I suggest writing route paths with lowercase and without whitespaces

**Comment by olivena** [comment]
> the ui - folder and it's content seems unnecessary, is it needed?

actually yes, i need to delete it

File: package-lock.json (hunk 1 above)
Comment (olivena) [reply package-lock.json]: where it should be placed?

Comment (rvsape) [reply package-lock.json]: in seems to have atleast the fonts, so inside the cc-ui. maybe undo changes of this file, and run the font installations inside the cc-ui folder?

Comment (rvsape) [reply package-lock.json]: or actually, it seems that the upper level does not have it's own package-json, so this file should be deleted and the dependencies (at least fonts?) should be installed inside the cc-ui folder

**Review by rvsape — CHANGES_REQUESTED** [review]
npm run dev error: "The following dependencies are imported but could not be resolved: @fontsource/montserrat"
- install font inside cc-ui
- delete package-lock.json from system root folder

File: cc-ui/src/pages/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+const FormLabel = chakra("text", {
+    baseStyle: {
+        float: 'left',
+        borderRadius: 'base',
+        color: '#EE0004'
+    }
+})
+*/
+export default function Login() {
+    const [username] = useState("");
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]:
```suggestion
    const [username, setUsername] = useState("");
    const [username, setPassword] = useState("");
```

File: cc-ui/src/pages/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                
+                    <FormLabel>Palveluntarjoaja</FormLabel>
+                    <Select borderColor='#EE0004' >
+                        <option>Palveluntarjoaja</option>
+                    </Select>
+
+                    <FormLabel>Käyttäjänimi</FormLabel>
+                    <Input type='string' 
+                        name='username'
+                        borderColor='#EE0004'
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: add onChange-handler, and update username
```suggestion
                     onChange={(e) => setUsername(e.target.value)}
```

Comment (rvsape) [reply cc-ui/src/pages/Login.tsx]: no need to delete the borderColor, I just dont know how to use this suggestion feature

File: cc-ui/src/pages/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    <Input type='string' 
+                        name='username'
+                        borderColor='#EE0004' 
+                    />
+                     
+                    <FormLabel>Salasana</FormLabel>
+                    <Input type={'password'} 
+                        name='password' 
+                        borderColor='#EE0004' 
+                    />
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: add onChange-handler, and update username
```suggestion
 onChange={(e) => setPassword(e.target.value)}   />
```

File: cc-ui/src/components/theme.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { extendTheme } from '@chakra-ui/react'
+import "@fontsource/montserrat"; 
+
+const theme = extendTheme({
+  fonts: {
+    heading: 'Montserrat',
+    body: 'Montserrat',
```
Comment (rvsape) [inline cc-ui/src/components/theme.tsx]: nice theming!

**Review by rvsape — CHANGES_REQUESTED** [review]

File: cc-ui/src/pages/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+                new URLSearchParams({
+                    'client_id': 'management',
+                    'grant_type': 'password',
+                    'username': username,
+                    'password': password
+                }));
+        return (await response).data.access_token;
+        } catch (err) {
+            console.log('Error: cannot fetch id for the current user');
+            var errorMessage = document.getElementById('wrongCred');
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: const/let instead of var

File: cc-ui/src/pages/Login.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+    
+    //Authentication to PK
+    async function handleSubmit(event: { preventDefault: () => void }) {
+        event.preventDefault();
+        console.log(username, password);
+
+        try {
+            const response = axios.post(
+                'https://auth.purkukartoitus.fi/auth/realms/rapurc/protocol/openid-connect/token',
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx:45]: authUrl variable can be used here instead the string

File: cc-ui/src/pages/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+                    />
+                     
+                    <FormLabel>Salasana</FormLabel>
+                    <Input type={'password'} 
+                        name='password' 
+                        borderColor='#EE0004' 
+                        onChange={(e) => setPassword(e.target.value)}
+                    />
+
+                    <FormLabel
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx:112]: useState for error, if true show this message
```suggestion
                    {formError && (<FormLabel stuff ..... />)}

```

File: cc-ui/src/pages/Login.tsx (hunk 7 above)
Comment (rvsape) [reply cc-ui/src/pages/Login.tsx]: also, useState with boolean

**Review by rvsape — CHANGES_REQUESTED** [review]

File: cc-ui/src/pages/Login.tsx [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+            errorMessage.hidden = false;
+            return '';
+        }
+    }
+
+    
+    
+    return (
+        <>
+        <Header></Header>
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: also, this Header should be rendered in App.tsx, inside the root div

**Comment by olivena** [comment]
I still need to create something to make sure users are all logged in to get in to other pages.

Edit: actually since this part is super hard for me could someone else take lead on this.

File: cc-ui/src/pages/Login.tsx [hunk 11]
```diff
@@ -109,21 +112,26 @@ export default function Login() {
...
+                        onClick={navigateHome}
                         >
                             Kirjaudu
                     </Button>
-                </Stack>
-            </form>
-        </Flex>
-        </>
+
+                    <Routes>
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: Routes are already declared at App.tsx so they should be removed

File: cc-ui/src/pages/Login.tsx [hunk 12]
```diff
@@ -109,21 +112,26 @@ export default function Login() {
...
                         >
-                            Virheellinen käyttäjätunnus tai salasana.</FormLabel>
+                            Virheellinen käyttäjätunnus tai salasana.
+                    </FormLabel>
 
                     <Button 
                         type="submit"
                         textTransform='uppercase'
                         colorScheme='blue'
+                        onClick={navigateHome}
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: Button type is submit, so onClick should trigger the onSubmit -function defined in the form element, so use that and remove this

File: cc-ui/src/pages/Login.tsx [hunk 13]
```diff
@@ -46,23 +50,22 @@ export default function Login() {
...
-            var errorMessage = document.getElementById('wrongCred');
-            errorMessage.hidden = false;
+            setWrongCredState(true);
             return '';
         }
     }
 
-    
-    
+    const navigateHome = () => {
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: remove this function

**Review by rvsape — CHANGES_REQUESTED** [review]

File: cc-ui/src/pages/Login.tsx [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+          id: 1,
+          displayValue: 'Purkukartoitus',
+          authEndpoint:
+            'https://auth.purkukartoitus.fi/auth/realms/rapurc/protocol/openid-connect/token',
+        },
+      ];
+
+    
+    //Authentication to PK
+    async function handleSubmit(event: { preventDefault: () => void }) {
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx:39]:
```suggestion
async function handleSubmit(event: { preventDefault: () => void }) {
    event.preventDefault();
    setWrongCredState(false);
    try {
      const response = await axios.post(
        'https://auth.purkukartoitus.fi/auth/realms/rapurc/protocol/openid-connect/token',
        new URLSearchParams({
          client_id: 'management',
          grant_type: 'password',
          username: username,
          password: password,
        })
      );
      console.log(response);
      if (response.status === 200) {
        setWrongCredState(false);
        navigate('/home');
      } else {
        throw new Error('login failed');
      }
    } catch (err) {
      setWrongCredState(true);
    }
  }
```

File: cc-ui/src/pages/Login.tsx [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+                })
+            );
+            console.log(response);
+            if ((await response).status === 200) {
+                setWrongCredState(false);
+                navigate('/home');
+              } else {
+                throw new Error('login failed');
+              }
+            return (await response).data.access_token;
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: no need for return in this function!

File: cc-ui/src/pages/Login.tsx [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+        } catch (err) {
+            setWrongCredState(true);
+            return '';
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: no need for return in this function!

**Review by rvsape — CHANGES_REQUESTED** [review]

File: cc-ui/src/pages/Login.tsx [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+                })
+            );
+            console.log(response);
+            if ((await response).status === 200) {
+                setWrongCredState(false);
+                navigate('/home');
+              } else {
+                throw new Error('login failed');
+              }
+            (await response).data.access_token;
```
Comment (rvsape) [inline cc-ui/src/pages/Login.tsx]: `(await response).data.access_token;`
remove this line

**Review by rvsape — CHANGES_REQUESTED** [review]
commented

**Review by rvsape — APPROVED** [review]

**Pull request merged**